### PR TITLE
Sils/langatts

### PR DESCRIPTION
**Pull request by sils1297**

### CODE REVIEW

**Comment by abhsag24** [comment]
ack 7ca3bc4

**Comment by abhsag24** [comment]
ack 1ce8166, 7b92d23

**Comment by Adrianzatreanu** [comment]
ack 7b92d2350

**Comment by sils1297** [comment]
rebase needed after merging a PR of @Adrianzatreanu

**Comment by sils1297** [comment]
reack 
https://github.com/coala-analyzer/coala-bears/pull/518/commits/aa699905ad9b33cb02273ba183c960726ca1e2e
https://github.com/coala-analyzer/coala-bears/pull/518/commits/fe816ab27867dfaae8b285e94f44671bf620596
3aed186

**Comment by sils1297** [comment]
reack aa69990 fe816ab

**Comment by Adrianzatreanu** [comment]
ack ae2a9d5c7813b

File: bears/c_languages/ClangBear.py [hunk 1]
```diff
@@ -27,6 +27,8 @@ class ClangBear(LocalBear):
     AUTHORS = {'The coala developers'}
     AUTHORS_EMAILS = {'coala-devel@googlegroups.com'}
     LICENSE = 'AGPL-3.0'
+    CAN_DETECT = {}
```
Comment (Adrianzatreanu) [inline bears/c_languages/ClangBear.py]: why is this needed? it is an empty set anyway. default

**Comment by Adrianzatreanu** [comment]
ack 785b098

File: bears/c_languages/ClangBear.py (hunk 1 above)
Comment (sils1297) [inline bears/c_languages/ClangBear.py]: I felt like explicitly setting that might be nice

**Comment by Adrianzatreanu** [comment]
ack 63d3409ff

File: bears/c_languages/CPPCleanBear.py [hunk 2]
```diff
@@ -19,6 +19,7 @@ class CPPCleanBear:
     AUTHORS = {'The coala developers'}
     AUTHORS_EMAILS = {'coala-devel@googlegroups.com'}
     LICENSE = 'AGPL-3.0'
+    CAN_DETECT = {'Smell', 'Redundancy', 'Security'}
```
Comment (Adrianzatreanu) [inline bears/c_languages/CPPCleanBear.py]: i know this may fit into Redundancy, but i think ``Unused Code`` should be mentioned. I used it a lot. unused code is sometimes different from redundancy

File: bears/c_languages/CPPLintBear.py [hunk 3]
```diff
@@ -20,6 +20,7 @@ class CPPLintBear:
     AUTHORS = {'The coala developers'}
     AUTHORS_EMAILS = {'coala-devel@googlegroups.com'}
     LICENSE = 'AGPL-3.0'
+    CAN_DETECT = {'Formatting'}
```
Comment (Adrianzatreanu) [inline bears/c_languages/CPPLintBear.py]: this catches style too, i googled it

**Comment by Adrianzatreanu** [comment]
ack f930cb1877c

**Comment by Adrianzatreanu** [comment]
ack f930cb18

**Comment by Adrianzatreanu** [comment]
ack c1819be37c9

**Comment by Adrianzatreanu** [comment]
ack 4c5a38929

File: bears/c_languages/ClangBear.py (hunk 1 above)
Comment (Adrianzatreanu) [inline bears/c_languages/ClangBear.py]: it feels misleading, as you already specify can fix, and it should have the same values at CAN_DETECT as well.

by the way you are pointing it, you are specifying explicitly that it cannot detect what it fixes, which we dont want to show

**Comment by Adrianzatreanu** [comment]
unack  588eaba3

**Comment by Adrianzatreanu** [comment]
ack e3b3a97ed

File: bears/c_languages/CPPCleanBear.py (hunk 2 above)
Comment (Adrianzatreanu) [inline bears/c_languages/CPPCleanBear.py]: we also no longer have Redundancy.

**Comment by Adrianzatreanu** [comment]
hey @sils1297 please dont forget about this if you want them for the release. Should go over the commits and remove ``Redundancy`` as we no longer use it, introduce ``Unused Code`` instead and check those 2 commits ive not acked too :) going to sleep now, reviewed some PRs and im kind of tired after a long day working

File: bears/c_languages/ClangBear.py [hunk 4]
```diff
@@ -27,6 +27,7 @@ class ClangBear(LocalBear):
     AUTHORS = {'The coala developers'}
     AUTHORS_EMAILS = {'coala-devel@googlegroups.com'}
     LICENSE = 'AGPL-3.0'
+    CAN_FIX = {'Variable Misuse', 'Syntax'}
```
Comment (hypothesist) [inline bears/c_languages/ClangBear.py]: I'd add `Semantics` (it's in the bear's description and it's a valid CAN_FIX value

**Comment by hypothesist** [comment]
ack https://github.com/coala-analyzer/coala-bears/pull/518/commits/aa33fc8702ba37363d876aa0e792415279465b0c https://github.com/coala-analyzer/coala-bears/pull/518/commits/5ebe7ed79547d95a8f9fc2a9c783530d57391a97 https://github.com/coala-analyzer/coala-bears/pull/518/commits/2b1d1dd0a5f2ea8a67ad40e13b9bd62a3ebbf2ea https://github.com/coala-analyzer/coala-bears/pull/518/commits/90dfbabc37d6e986d931aa5e849a7d285ed043cf

**Comment by hypothesist** [comment]
@rultor merge
